File: app/src/main/java/com/legue/axel/lolsummonertool/database/viewmodel/ItemGoldViewModel.java
package com.legue.axel.lolsummonertool.database.viewmodel;

import android.app.Application;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.annotation.NonNull;

import com.legue.axel.lolsummonertool.database.SummonerToolDatabase;
import com.legue.axel.lolsummonertool.database.model.item.ItemGold;

public class ItemGoldViewModel extends AndroidViewModel {

    private SummonerToolDatabase database;

    public ItemGoldViewModel(@NonNull Application application) {
        super(application);
        database = SummonerToolDatabase.Companion.getInstance(this.getApplication());
    }

    public LiveData<ItemGold> getItemGoldByItemId(int itemId) {
        return database.itemGoldDao().getItemGoldByItemId(itemId);
    }
}

File: app/src/main/java/com/legue/axel/lolsummonertool/viewmodel/ItemGoldViewModel.kt
package com.legue.axel.lolsummonertool.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

import com.legue.axel.lolsummonertool.database.SummonerToolDatabase
import com.legue.axel.lolsummonertool.database.model.item.ItemGold

class ItemGoldViewModel(application: Application) : AndroidViewModel(application) {

    private val database: SummonerToolDatabase = SummonerToolDatabase.getInstance(this.getApplication())

    fun getItemGoldByItemId(itemId: Int): LiveData<ItemGold> {
        return database.itemGoldDao().getItemGoldByItemId(itemId)
    }
}

